app/workers/publish_data_worker.rb

```diff
@@ -0,0 +1,16 @@
+class PublishDataWorker
```

Comment:
This seems a bit excessive. Without benchmarks, I don't really believe looking up the classification 4 times (lifecycle, this worker, two workers queued by this one) will actually be faster than inserting two items in the queue from the lifecycle worker and losing one classification lookup.